File: Java/Easy/RomanToInt.java
//Problem's link: https://leetcode.com/problems/roman-to-integer/

public int romanToInt(String s) {
	int result = 0;
	for(int i = s.length()-1 ; i >= 0 ; i--) {
		final var current = s.charAt(i);
		final var previous = i != s.length()-1 ? s.charAt(i+1) : current;
		switch(current) {
			case 'I':
				result += previous == 'V' || previous == 'X' ? -1: 1;
				break;
			case 'V':
				result += 5;
				break;
			case 'X':
				result += previous == 'C' || previous == 'L' ? -10: 10;
				break;
			case 'L':
				result += 50;
				break;
			case 'C':
				result += previous == 'D' || previous == 'M' ? -100: 100;
				break;
			case 'D':
				result += 500;
				break;
			case 'M':
				result += 1000;
				break;
		}
	}

	return result;
}

File: Kotlin/Easy/RomanToInt.kt
//Problem's link: https://leetcode.com/problems/roman-to-integer/

fun romanToInt(s: String): Int {
    var res = 0
    for(index in s.indices.reversed()) {
        val current = s[index]
        val previous = if(index != s.length -1) s[index + 1] else current
        when(current) {
            'I' -> res += if(previous == 'V' || previous == 'X') -1 else 1
            'V' -> res += 5
            'X' -> res += if(previous == 'L' || previous == 'C') -10 else 10
            'L' -> res += 50
            'C' -> res += if(previous == 'D' || previous == 'M') -100 else 100
            'D' -> res += 500
            'M' -> res += 1000
        }
    }
    return res
}